formats/qcow-efi.nix

```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, modulesPath, ... }:
+{
+  # for virtio kernel drivers
+  imports = [
+    "${toString modulesPath}/profiles/qemu-guest.nix"
+  ];
+
+  fileSystems."/" = {
+    device = "/dev/disk/by-label/nixos";
+    autoResize = true;
+    fsType = "ext4";
+  };
+
+  boot.growPartition = true;
+  boot.kernelParams = [ "console=ttyS0" ];
+  boot.loader.grub.device = "nodev";
+
+  boot.loader.grub.efiSupport = true;
+  boot.loader.grub.efiInstallAsRemovable = true;
+  boot.loader.timeout = 0;
+
+  system.build.qcow = import "${toString modulesPath}/../lib/make-disk-image.nix" {
+    inherit lib config pkgs;
+    diskSize = 2048;
```

Comment:
Should this be configurable?

Reply:
how to make it configurable

Reply:
By using `mkOption` to create an options in this module and than default this option to 2048. This is an example how to define options: https://github.com/NixOS/nixpkgs/blob/dbf4ff8fbf618ffca869cf4f82d83dfc61ed4eb8/nixos/modules/services/networking/ferm.nix#L30